staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/sort.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type managedFieldsLessFunc func(p, q metav1.ManagedFieldsEntry) bool
+
+type managedFieldsSorter struct {
+	fields []metav1.ManagedFieldsEntry
+	less   []managedFieldsLessFunc
+}
+
+func (s *managedFieldsSorter) Sort(fields []metav1.ManagedFieldsEntry) {
+	s.fields = fields
+	sort.Sort(s)
+}
+
+// Len is the amount of managedFields to sort
+func (s *managedFieldsSorter) Len() int {
```

Comment:
I suspect you can use [Slice](https://golang.org/pkg/sort/#Slice) to avoid the boilerplate.

Reply:
Right, that's much cleaner now. With other projects I had to keep compatibility before go1.8 so Slice wasn't an option and I got used to doing it that way.